### PR TITLE
Upgrade projects and supporting files to ASP.NET Core 2.1

### PR DESCRIPTION
**Pull request by rafsanulhasan**
1. Updated the whole solution in compliance with **.NET Core 2.1**.
2. Underlying projects now uses **ASP.NET Core 2.1**. 
3. All the startup and bootstrap file (program) for _API/Web/SPA/Gateway(Aggregator)_ projects are changed accordingly. 
a. In whichever projects it is applicable in _Startup.cs_, it uses _.SetCompatibilityVersion_ to **ASP.NET core 2.1**
b. In whichever projects it is applicable in _Startup.cs_, it uses **_GDPR Cookie Compliance_** 
4. All the supporting files including dockerfiles, launch settings, and other json files, code files (cs) and Razor cshtml files are updated from **"net21rc1"** branch

### CODE REVIEW

**Comment by rafsanulhasan** [comment]
@CESARDELATORRE @davidfowl

File: src/ApiGateways/ApiGw-Base/Program.cs [hunk 1]
```diff
@@ -20,11 +20,11 @@ public static void Main(string[] args)
 
         public static IWebHost BuildWebHost(string[] args)
         {
-            var builder = WebHost.CreateDefaultBuilder(args);
```
Comment (davidfowl) [inline src/ApiGateways/ApiGw-Base/Program.cs]: Why did you remove var?

File: src/ApiGateways/Mobile.Bff.Shopping/aggregator/Services/CatalogService.cs [hunk 2]
```diff
@@ -1,43 +1,41 @@
...
+
+            return catalogItem;
         }
 
         public async Task<IEnumerable<CatalogItem>> GetCatalogItems(IEnumerable<int> ids)
         {
-            var data = await _apiClient.GetStringAsync(_urls.Catalog + UrlsConfig.CatalogOperations.GetItemsById(ids));
-            var item = JsonConvert.DeserializeObject<CatalogItem[]>(data);
-            return item;
+            var stringContent = await _httpClient.GetStringAsync(_urls.Catalog + UrlsConfig.CatalogOperations.GetItemsById(ids));
```
Comment (davidfowl) [inline src/ApiGateways/Mobile.Bff.Shopping/aggregator/Services/CatalogService.cs]: While you're at it, fix these calls by using the GetJsonAsync methods in the https://www.nuget.org/packages/Microsoft.AspNet.WebApi.Client/ package.

File: src/ApiGateways/Mobile.Bff.Shopping/aggregator/Startup.cs [hunk 3]
```diff
@@ -102,37 +138,47 @@ public void ConfigureServices(IServiceCollection services)
...
 
-        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
-        public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
+            return services;
+        }
+        public static IServiceCollection AddHttpServices(this IServiceCollection services)
         {
+            //register delegating handlers
+            services.AddTransient<HttpClientAuthorizationDelegatingHandler>();
+            services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
```
Comment (davidfowl) [inline src/ApiGateways/Mobile.Bff.Shopping/aggregator/Startup.cs]: services.AddHttpContextAccessor();

File: src/ApiGateways/Mobile.Bff.Shopping/aggregator/Services/CatalogService.cs (hunk 2 above)
Comment (rafsanulhasan) [reply src/ApiGateways/Mobile.Bff.Shopping/aggregator/Services/CatalogService.cs]: I cannot use this package in a asp.net core project since it is targeted for ASP.NET.
The best I can do is below:
```var response = await _httpClient.GetAsync(_urls.Basket + UrlsConfig.BasketOperations.GetItemById(id));```
```BasketData basket = response.IsSuccessStatusCode ? await response.Content.ReadAsAsync<BasketData>() : null;```

File: src/ApiGateways/Mobile.Bff.Shopping/aggregator/Startup.cs (hunk 3 above)
Comment (rafsanulhasan) [reply src/ApiGateways/Mobile.Bff.Shopping/aggregator/Startup.cs]: Done

File: src/ApiGateways/ApiGw-Base/Program.cs (hunk 1 above)
Comment (rafsanulhasan) [reply src/ApiGateways/ApiGw-Base/Program.cs]: Reverted

File: src/ApiGateways/Mobile.Bff.Shopping/aggregator/Services/CatalogService.cs (hunk 2 above)
Comment (rafsanulhasan) [reply src/ApiGateways/Mobile.Bff.Shopping/aggregator/Services/CatalogService.cs]: Besides, the author of the project allowed this operations are invoked like that. So I think nothing to do with that. What do you think @CESARDELATORRE

**Comment by mvelosop** [comment]
Hi @rafsanulhasan, thanks for the PR

I'm a bit confused here!

This PR is to be merged into branch features/netcore2.1 which was merged into dev on MAY-11 and is closed now.

Also some of the changes included have been committed to dev since MAY-15, so it looks like this PR is based on an outdated branch.

OTOH reviewing such a massive PR is quite difficult so PRs should be focused regarding functionality and the microservices affected.

So you should always begin a PR by creating your branch from the latest dev in this repo and pick a small set of changes.

And please, check with @CESARDELATORRE if you intent to invest a significant effort in some changes, just to make sure whatever you want to do is aligned with the current priorities.

So I'm closing this PR and continue the conversation in some other issue or PR.
